src/uwtools/config.py

```diff
@@ -301,6 +301,30 @@ def update_values(self, new_dict, dict_to_update=None):
             else:
                 dict_to_update[key] = new_val
 
+    def iterate_values(self, config_dict, set_var, jinja2_var, undeclared_var):
+        '''
+        Given a config dictionary, parse which variables have been set, which are left as 
+        jinja2 templates, and which are undeclared
+        '''
```

Comment:
It might be helpful to add some information about the arguments, including hints that the last three are lists.

Reply:
Suggest an edit:
```
        Recursively parse which keys in the object are complete (set_var), which keys have 
        unfilled jinja templates (jinja2_var), and which keys are set to empty (empty_var).  
```